bin/copyProjectDataToPrm.sh

```diff
@@ -345,6 +346,52 @@ function archiveSampleSheet() {
 	fi
 }
 
+function getSampleType(){
+	local  _sampleSheet="${1}"
+	declare -a sampleSheetColumnNames=()
+	declare -A sampleSheetColumnOffsets=()
+	declare    sampleType='DNA' # Default when not specified in sample sheet.
+	declare    sampleTypeFieldIndex
+	IFS="${SAMPLESHEET_SEP}" sampleSheetColumnNames=($(head -1 "${_sampleSheet}"))
+	for (( offset = 0 ; offset < ${#sampleSheetColumnNames[@]:-0} ; offset++ ))
+	do
+		#
+		# Backwards compatibility for "Sample Type" including - the horror - a space and optionally quotes :o.
+		#
+		regex='Sample Type'
+		if [[ "${sampleSheetColumnNames[${offset}]}" =~ ${regex} ]]
+		then
+			columnName='sampleType'
+		else
+			columnName="${sampleSheetColumnNames[${offset}]}"
+		fi
+		sampleSheetColumnOffsets["${columnName}"]="${offset}"
+		log4Bash 'DEBUG' "${LINENO}" "${FUNCNAME:-main}" '0' "${columnName} and sampleSheetColumnOffsets["${columnName}"] offset ${offset} "
+	done
+	
+	if [[ ! -z "${sampleSheetColumnOffsets['sampleType']+isset}" ]]; then
+		#
+		# Get sampleType from sample sheet and check if all samples are of the same type.
+		#
+		sampleTypeFieldIndex=$((${sampleSheetColumnOffsets['sampleType']} + 1))
+		sampleTypesCount=$(tail -n +2 "${_sampleSheet}" | cut -d "${SAMPLESHEET_SEP}" -f ${sampleTypeFieldIndex} | sort | uniq | wc -l)
+		if [[ "${sampleTypesCount}" -eq '1' ]]
+		then
+			sampleType=$(tail -n 1 "${_sampleSheet}" | cut -d "${SAMPLESHEET_SEP}" -f ${sampleTypeFieldIndex})
```

Comment:
${sampleTypeFieldIndex} Hier ook dubbele quotes toevoegen? :-)

Reply:
quotes toegevoegd aan trace and trace ;)